specs/beacon-chain.md

```diff
@@ -742,11 +743,15 @@ For every shard number `shard` for which a crosslink committee exists in the cyc
 
 #### Process penalties, logouts and other special objects
 
+Verify that there are at most `MAX_SPECIALS_PER_BLOCK` special records.
```

Comment:
This should go up in the block validity conditions rather than in the state recalc section

Reply:
`MAX_SPECIALS_PER_BLOCK` feels suboptimal because specials have different overhead for beacon chain. Here are two extreme scenarios:

1) The block proposers have a large influx of lightweight specials where including more than `MAX_SPECIALS_PER_BLOCK` per block would be preferable.
2) An attacking proposer DoSes the network by filling blocks with `MAX_SPECIALS_PER_BLOCK` maximally-heavy specials.

What about replacing total specials count by total specials size? (E.g. a maximum of 4kB of specials per block.) Total specials size only captures the bandwidth overhead (not the computational overhead), so maybe it's wise to have both total count and total size limits.

Reply:
Having a max specials per block along with a block size limit might be the right compromise

Reply:
I'm ok either way. Maybe better to not touch this until we're done all the specials including proof of custody challenges, and we get numbers on how long each take to process?